### PR TITLE
100% Type Safety & Python3.11 Upgrade

### PR DESCRIPTION
**Pull request by cmyui**

### CODE REVIEW

File: app/api/direct.py [hunk 1]
```diff
@@ -130,18 +131,18 @@ async def osu_direct(
...
     user: User = Depends(authenticate_user(Query, "u", "h")),
     map_set_id: Optional[int] = Query(None, alias="s"),
     map_id: Optional[int] = Query(None, alias="b"),
-):
+) -> Response:
     if map_set_id is None and map_id is not None:
         bmap = await app.usecases.beatmap.fetch_by_id(map_id)
         if bmap is None:
-            return
+            return Response(b"")
```
Comment (cmyui) [inline app/api/direct.py:145]: Behaviour change with these -- from `b"none"` to `b""` response data

File: app/api/pp.py [hunk 2]
```diff
@@ -26,12 +27,10 @@ async def calculate_pp(
     mode_arg: int = Query(0, alias="g", ge=0, le=3),
     acc: Optional[float] = Query(None, alias="a"),
     combo: int = Query(0, alias="max_combo"),
-):
+) -> Response:
     mods = Mods(mods_arg)
     mode = Mode.from_lb(mode_arg, mods_arg)
 
-    use_common_pp_percentages = acc is None
```
Comment (cmyui) [inline app/api/pp.py:33]: Moved this to inline so mypy can more quickly determine `acc`'s type later on

File: app/usecases/countries.py [hunk 3]
```diff
@@ -4,12 +4,8 @@
 
 
 async def get_country(user_id: int) -> str:
-    country = await app.state.services.database.fetch_val(
+    country: str = await app.state.services.database.fetch_val(
```
Comment (tsunyoku) [inline app/usecases/countries.py]:
```suggestion
    country: str | None = await app.state.services.database.fetch_val(
```

File: app/usecases/pp_cap.py [hunk 4]
```diff
@@ -15,7 +15,7 @@ async def get_pp_cap(mode: Mode, mods: Mods) -> int:
     if mods & Mods.FLASHLIGHT:
         prefix += "flashlight_"
 
-    pp_cap = await app.state.services.database.fetch_val(
+    pp_cap: int = await app.state.services.database.fetch_val(
```
Comment (tsunyoku) [inline app/usecases/pp_cap.py]:
```suggestion
    pp_cap: int | None = await app.state.services.database.fetch_val(
```

**Review by tsunyoku — CHANGES_REQUESTED** [review]

**Review by tsunyoku — APPROVED** [review]

**Pull request merged**